cc-service-impl/src/main/java/com/jtmc/apps/reforma/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.jtmc.apps.reforma.repository;
+
+import com.google.inject.Inject;
+import com.jtmc.apps.reforma.domain.Login;
+import com.jtmc.apps.reforma.domain.PersonaDetails;
+import com.jtmc.apps.reforma.domain.UserDetails;
+import com.jtmc.apps.reforma.repository.mapper.*;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.mybatis.dynamic.sql.BasicColumn;
+import org.mybatis.dynamic.sql.select.render.SelectStatementProvider;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.mybatis.dynamic.sql.SqlBuilder.isEqualTo;
+import static org.mybatis.dynamic.sql.SqlBuilder.isTrue;
+import static org.mybatis.dynamic.sql.util.mybatis3.MyBatis3Utils.select;
+
+public class UserRepositoryImpl {
+
+    @Inject
+    private SqlSessionFactory sqlSessionFactory;
+
+    public UserDetails selectUser(String username) {
+        UserDetails userDetails = new UserDetails();
+        // branchId
+        Optional<Login> loggedInUser = this.selectLoggedInUser(username);
+        if(!loggedInUser.isPresent()) {
+            throw new RuntimeException("No valid LoggedInUser");
+        }
+        // username
+        userDetails.setUsername(username);
+        // roles
+        Collection<PersonaDetails> personaDetails = this.selectUserRoles(loggedInUser.get().getPersonaid());
+        Stream<Integer> rolesId = personaDetails.stream().map(PersonaDetails::getPersonaid);
+        List<Integer> roles = rolesId.collect(Collectors.toList());
+        userDetails.setRoles(roles);
+
+        //todo: for know default branch will be the first value
+        // for sure I know only one row per user will be registered in persona_details
+        // version 0.5.0
+        Optional<PersonaDetails> selectingBranch = personaDetails.stream().findFirst();
+        userDetails.setDefaultBranch(selectingBranch.isPresent() ? selectingBranch.get().getBranchid() : 0);
+        return userDetails;
```

Comment:
//todo: for now default branch will be the first value
        // for sure I know only one row per user will be registered in persona_details
        // version 0.5.0